app/actions/handlers.py

```
@@ -0,0 +1,191 @@
+        async for attempt in stamina.retry_context(
+                on=httpx.HTTPError,
+                attempts=3,
+                wait_initial=datetime.timedelta(seconds=10),
+                wait_max=datetime.timedelta(seconds=10),
+        ):
+            with attempt:
+                vehicles = await client.get_positions_list(
+                    integration=integration,
+                    config=action_config
+                )
+
+        if vehicles:
+            logger.info(f"Observations pulled with success. Length: {len(vehicles)}")
+
+            transformed_data = await filter_and_transform(
+                vehicles,
+                str(integration.id),
+                "pull_observations"
+            )
+
+            if transformed_data:
+                async for attempt in stamina.retry_context(
+                        on=httpx.HTTPError,
+                        attempts=3,
+                        wait_initial=datetime.timedelta(seconds=10),
+                        wait_max=datetime.timedelta(seconds=10),
+                ):
+                    with attempt:
+                        try:
+                            response = await send_observations_to_gundi(
+                                observations=transformed_data,
+                                integration_id=str(integration.id)
+                            )
+                        except httpx.HTTPError as e:
+                            msg = f'Sensors API returned error for integration_id: {str(integration.id)}. Exception: {e}'
+                            logger.exception(
+                                msg,
+                                extra={
+                                    'needs_attention': True,
+                                    'integration_id': str(integration.id),
+                                    'action_id': "pull_observations"
+                                }
+                            )
+                            return [msg]
```

If we silence the exception here then we will be hiding errors. We will see all good in activity logs and then somewhere in the details we will see the error message. I'd prefer that we raise the exception so we see the error in the activity logs.